### PR TITLE
fix(overlay-menu): missing nested items property

### PR DESCRIPTION
**Pull request by Nantawat-Poothong**
## Description
Typescript compiler show type error when trying to set value with a nested data.

Root cause:
-  Typescript compiler not understand Discriminated union types with nested. So it will guest one of a union types
For example:  https://codesandbox.io/s/nested-discriminated-unions-original-hlm5h3?file=/src/index.ts

Investigation result: 
- Typescript don't understand `this[];` when using with Discriminated union types
- After investigate such a long time, one thing we know well is that nested `items?: this[];` is only use by OverlayMenu, so  it should be fine to change from `this[]` to  something else, as long as OverlayMenu type  not broken.

Solution:
- Change CollectionItem `items?: this[]` to support recursive them self `items?: CollectionItem[]` and let CommonItem support items type `ItemData[]`;
For example: https://codesandbox.io/s/nested-discriminated-unions-ap2-2fcnwr?file=/src/index.ts

Why this solution work?
this example will show that the `pet` property of `Home` type can be `Dog` type instead of `Pet` type as long as it extend from `Pet` type

```ts
interface Pet {
  name: string;
}

interface Dog extends Pet {
  sign: string;
}

interface House {
  pet?: Pet;
}

interface Home extends House {
  pet?: Dog;
}
```

Fixes # (issue)
https://jira.refinitiv.com/browse/STG-191

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/utils/src/collection/collection-item.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ export interface CollectionItem {
   /**
    * Child items collection.
    * Used for nested data structures.
+   * Override this property in sub-interface type to ensure nested type checking
```
Comment (wattachai-lseg) [inline packages/utils/src/collection/collection-item.ts]:
```suggestion
   * Override this property in sub interfaces with their own type to ensure nested type checking.
   * For example, `MyCollectionItem extends CollectionItem { items?: MyCollectionItem[]; }`
```

File: documents/src/pages/utils/data-management.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ interface CollectionItem {
   /**
    * Child items collection.
    * Used for nested data structures.
+   * Override this property in sub-interface type to ensure nested type checking
```
Comment (wattachai-lseg) [inline documents/src/pages/utils/data-management.md]:
```suggestion
   * Override this property in sub interfaces with their own type to ensure nested type checking.
   * For example, `MyCollectionItem extends CollectionItem { items?: MyCollectionItem[]; }`
```

**Review by wattachai-lseg — APPROVED** [review]

**Comment by Theeraphat-Sorasetsakul** [comment]
I did verify and found this. To set only label or value to nested item, it throws type error. Is this expected?
```javascript
overlaymenu.data = [{
  label: 'item1',
  items: [{
    label: "item1.1",
    // Type '{ label: string; }' is not assignable to type 'ItemData'.
    // Property 'type' is missing in type '{ label: string; }' but required in type 'ItemDivider'.
    items: [{
      label: "item1.1.1",
    }],
  }]
}]
```

**Comment by Nantawat-Poothong** [comment]
> I did verify and found this. To set only label or value to nested item, it throws type error. Is this expected?
> 
> ```js
> overlaymenu.data = [{
>   label: 'item1',
>   items: [{
>     label: "item1.1",
>     // Type '{ label: string; }' is not assignable to type 'ItemData'.
>     // Property 'type' is missing in type '{ label: string; }' but required in type 'ItemDivider'.
>     items: [{
>       label: "item1.1.1",
>     }],
>   }]
> }]
> ```
This is expected, it show an error because the data didn't match with the interface.

```typescript
interface CommonLabelItem extends CommonItem {
  label: string; // required
  icon?: string;
  tooltip?: string;
}

interface ItemText extends CommonLabelItem {
  type?: 'text';
  value: string;  // required
  subLabel?: string;
  for?: string;
}
```

**Comment by Theeraphat-Sorasetsakul** [comment]
Verified.

**Pull request merged**